ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -490,6 +469,42 @@ export const ApplicationResourceTree = (props: ApplicationResourceTreeProps) =>
     let roots: ResourceTreeNode[] = [];
     const childrenByParentKey = new Map<string, ResourceTreeNode[]>();
 
+    const [filters, setFilters] = React.useState(props.filters);
+    const [filteredGraph, setFilteredGraph] = React.useState([]);
+    const filteredNodes: any[] = [];
```

Comment:
Is there already a type for this kind of `node`? Just wondered why it has to be `any` type.

Reply:
The type that is used for filter is `ResourceTreeNode` but there are a few type castings that happen on the graph side of it, that is why I kept it as any. I can change it to the correct type if it is preferable.

Reply:
Ok, then you can leave it as `any`. When we have a certain type then better use the specific type than `any`.